synfig-studio/src/synfigapp/actions/layerduplicate.cpp

```diff
@@ -418,21 +357,13 @@ LayerDuplicate::replace_valuenodes(const std::map<synfig::Layer::Handle,synfig::
 	for (const auto& layer_pair : cloned_layer_map) {
 
 		auto cloned_layer = layer_pair.second;
-
-		// disconnect_dynamic_param/connect_dynamic_param can change dynamic_param_list() while iterating
-		// which makes iter invalid, so we create a copy of dynamic_param_list() first
-		auto param_list = cloned_layer->dynamic_param_list();
-
-		for (auto iter=param_list.cbegin();iter!=param_list.cend();++iter)
-		{
-			for (const auto& vn_pair : cloned_valuenode_map) {
-				if (iter->second == vn_pair.first) {
-					cloned_layer->disconnect_dynamic_param(iter->first);
-					cloned_layer->connect_dynamic_param(iter->first, vn_pair.second);
-				} else {
-					do_replace_valuenodes(iter->second, vn_pair);
-				}
+		replace_value_nodes(cloned_layer,
+							[cloned_valuenode_map](ValueNode::LooseHandle vn) -> ValueNode::LooseHandle {
```

Comment:
Why not use `ValueNode::LooseHandle& vn` here?

Reply:
Well, I'm not sure :D 

1. To explicitly know if some replacement is available (we could return bool though)
2. We don't necessarily have to replace the item itself
See synfig_iterations.cpp line 150.

I think  it's safer this way.

Reply:
I mean, `vn` is only used for map lookup here. So you can replace it with `[cloned_valuenode_map] (const ValueNode::LooseHandle& vn)` without losing any functionality. The only thing that will change is it will not copy `vn` on every iteration. Or I miss something?

Reply:
As `LooseHandle` is just a "pointer", I thought it wouldn't be a problem.

Reply:
Done